### ISSUE TITLE
linter: oxlint-plugin-eslint incorrectly reports unused variables

### ISSUE THREAD

**Issue by KTrain5169** [oxc-project/oxc#21520]
### What version of Oxlint are you using?

1.60.0

### What command did you run?

`oxlint --type-aware` & `oxc.oxc-vscode` VS Code extension

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
// minimal config to reproduce
{
    "jsPlugins": ["oxlint-plugin-eslint"],
    "rules": {
        "eslint-js/no-unused-vars": "errors"
    }
}
```

### What happened?

`const enum` members and input variables in function typings are considered "unused", despite:

1. all `const enum` members actually being used
2. the function typing being just that - a type, not an actual function

Below are screenshots from my editor:

<img width="1736" height="340" alt="Image" src="https://github.com/user-attachments/assets/e98b4bcf-7f56-497f-85cc-ee82e8230b8b" />
This screenshot demontrates that const enum members, despite being used (even within the same file!), are labeled as unused by the plugin.

<img width="867" height="193" alt="Image" src="https://github.com/user-attachments/assets/0b35e666-5d3c-4de2-ba6f-faed9e2ab20b" />
This screenshot demonstrates that function typing shows an error with the above configuration.

**Comment by camc314** [comment]
Hi, this is working as expected. eslint-js/no-unused-vars does not support typescript syntax.

In order to support this, please swap out eslint-js for `@typescript-eslint/eslint-plugin` and use the typescript-eslint version of this rule.


Please reach out with any issues!


Thanks!
